### PR TITLE
RFC42 style knot config

### PR DESCRIPTION
**Pull request by lheckemann**
###### Motivation for this change
An alternative approach to #81460 (closes #81460), using RFC42 style (defining a format) and with tests on the format.

Currently blocked on the test failing (see commit message de5b12a988e72fa6d4241f2d1d5c11134ab21acd for details)

cc co-authors @elseym, @Ma27

### CODE REVIEW

File: pkgs/pkgs-lib/formats.nix [hunk 1]
```diff
@@ -128,4 +128,77 @@ rec {
...
+    - Keys may be duplicated, in which case the resulting value is a
+      list composed of all the values set (which we use here)
+
+    - The parser cares a great deal about the order of keys: for
+      instance, when defining an ACL, the `id` attribute must be the
+      first. Unfortunately, this is not documented, nor is the error
+      message thrown if these conditions aren't fulfilled very
+      helpful.
+
+   … so we have an extra format just for knot.
```
Comment (mweinelt) [inline pkgs/pkgs-lib/formats.nix:148]: Yay. :disappointed:

**Comment by vcunat** [comment]
I only had a quick look so far, but I don't see it ordering the sections.  Explanation from my draft PR:
>             # We output the config section in the upstream-mandated order.
>             # Ordering is important due to forward-references not being allowed.

**Comment by sinavir** [comment]
Hello, any updates on this ? How can I help ?

It seems there are two milestones:

- [ ] Handling ordering of sections
- [ ] The check-conf logic

The second point seems to me a bit not related to the PR since including keys from outside the store is also broken in the current module. I believe it isn't possible to make a full check at build-time thus I think we should remove the check.

**Comment by vcunat** [comment]
I assume this PR is deprecated now.  Feel free to improve the alternative that we have there now.

**Pull request closed without merging**